File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void postorderTraversal_01(TreeNode* node,vector<int>& ans) {
        //Base Case
        if(node == nullptr) return;
        
        //Left Recursive Call
        postorderTraversal_01(node->left,ans);
        // INORDER ------>>>>
        // INORDER ------>>>>
        //Right Recursive Call
        postorderTraversal_01(node->right,ans);
        
         ans.push_back(node->val);
    }
    vector<int> postorderTraversal(TreeNode* root) {
        vector<int> ans;
        postorderTraversal_01(root,ans);
        return ans;
    }
};

File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public void postorderTraversal_01(TreeNode node,ArrayList<Integer> ans) {

        if(node.left != null){
            postorderTraversal_01(node.left,ans);
        }
        
        if(node.right != null){
            postorderTraversal_01(node.right,ans);
         }
        ans.add(node.val);
    }
    
    public List<Integer> postorderTraversal(TreeNode root) {
        ArrayList<Integer> ans = new ArrayList<>();
        if(root == null) return ans;
        postorderTraversal_01(root,ans);
        return ans;
    }
}